src/clients/TransactionClient.ts

```diff
@@ -23,6 +24,16 @@ export class TransactionClient {
   // eslint-disable-next-line no-useless-constructor
   constructor(readonly logger: winston.Logger) {}
 
+  protected async _simulate(txn: AugmentedTransaction): Promise<TransactionSimulationResult> {
+    return await willSucceed(txn);
+  }
+
+  // Each transaction is simulated in isolation; but on-chain execution may produce different
+  // results due to execution sequence or intermediate changes in on-chain state.
+  async simulate(txns: AugmentedTransaction[]): Promise<TransactionSimulationResult[]> {
+    return await Promise.all(txns.map((txn: AugmentedTransaction) => this._simulate(txn)));
```

Comment:
Can ts not infer the AugmentedTransaction type in the inner function?

```suggestion
 simulate(txns: AugmentedTransaction[]): Promise<TransactionSimulationResult[]> {
    return Promise.all(txns.map((txn: AugmentedTransaction) => this._simulate(txn)));
```